Ability to global enable/disable plugin (so that can be used for different envs)

## Expected Behavior
Want to be able to disable plugin activity (function rewriting etc) for certain envs (eg. personal stacks)

## Actual Behavior
Currently, having this kind of config results in a successful deployment via `sls deploy` but functions fail on execution.

```
dataDogForwarderLambdaArn:
    none: ''
    staging: arn:aws:lambda:us-east-1:blah:function:datadog-ForwarderStack-blah-Forwarder-blah
    prod: arn:aws:lambda:us-east-1:blah:function:datadog-ForwarderStack-blah-Forwarder-blah
  datadogEnabled:
    dev: false
    staging: true
    prod: true
  datadog:
    enableDDTracing: ${self:custom.datadogEnabled.${opt:stackName, 'dev'}, false}
    addLayers: ${self:custom.datadogEnabled.${opt:stackName, 'dev'}, false}
    forwarder: ${self:custom.dataDogForwarderLambdaArn.${opt:stackName, 'none'}, ''}
```
Functions fail with:
```
"errorType": "Runtime.ImportModuleError",
    "errorMessage": "Error: Cannot find module 'handler'\nRequire stack:\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",
    "stack": [
        "Runtime.ImportModuleError: Error: Cannot find module 'handler'",
        "Require stack:",
        "- /var/runtime/UserFunction.js",
        "- /var/runtime/index.js",
        "    at _loadUserApp (/var/runtime/UserFunction.js:100:13)",
        "    at Object.module.exports.load (/var/runtime/UserFunction.js:140:17)",
        "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
        "    at Module._compile (internal/modules/cjs/loader.js:1015:30)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)",
        "    at Module.load (internal/modules/cjs/loader.js:879:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:724:14)",
        "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)",
        "    at internal/main/run_main_module.js:17:47"
    ]
```

which I suspect is to do with the plugin "half" doing it's job, eg. having a forwarder with value '' ?

## Steps to Reproduce the Problem

  1. 
  1.
  1.

## Specifications

  - Serverless Framework version: 2.11.1
  - Datadog Serverless Plugin version: 2.10.1
  - Lambda function runtime (Python 3.7, Node 10, etc.):  Node 12.19
  
Note, plugin works as expected for `staging` & `prod` stackNames in this config.

--- a/README.md
+++ b/README.md
@@ -35,7 +35,7 @@ To further configure your plugin, use the following custom parameters in your `s
 | `forwarder`          | Setting this parameter subscribes the Lambda functions' CloudWatch log groups to the given Datadog forwarder Lambda function. Required when `enableDDTracing` is set to `true`.                                                                                                                                                                                                                 |
 | `enableTags`         | When set, automatically tag the Lambda functions with the `service` and `env` tags using the `service` and `stage` values from the serverless application definition. It does NOT override if a `service` or `env` tag already exists. Defaults to `true`.                                                                                                                                      |
 | `injectLogContext`         | When set, the lambda layer will automatically patch console.log with Datadog's tracing ids. Defaults to `true`.                                                                                                                                      |
-| `enabled`            | When set to false, the DataDog plugin will stay inactive. Defaults to `true`. You can control this option using an environment variable, e.g. `enabled: ${strToBool(${env:DD_PLUGIN_ENABLED, true})}`, to activate/deactivate the plugin during deployment. |
+| `enabled`            | When set to false, the DataDog plugin will stay inactive. Defaults to `true`. You can control this option using an environment variable, e.g. `enabled: ${strToBool(${env:DD_PLUGIN_ENABLED, true})}`, to activate/deactivate the plugin during deployment. Alernatively, you can also use the value passed in through `--stage` to control this option, [see example.](#disable-plugin-for-particular-environment)|
 
 
 To use any of these parameters, add a `custom` > `datadog` section to your `serverless.yml` similar to this example:
@@ -108,6 +108,24 @@ If you run into the following error, double check the supplied Forwarder ARN is 
 An error occurred: GetaccountapiLogGroupSubscription - Could not execute the lambda function. Make sure you have given CloudWatch Logs permission to execute your function. (Service: AWSLogs; Status Code: 400; Error Code: InvalidParameterException).
 ```
 
+### Disable Plugin for Particular Environment
+
+If you'd like to turn off the plugin based on the environment (passed via `--stage`), you can use something similar to the example below.
+
+```
+provider:
+  stage: ${self:opt.stage, 'dev'}
+
+custom:
+  staged: ${self:custom.stageVars.${self:provider.stage}, {}}
+
+  stageVars:
+    dev:
+      dd_enabled: false
+
+  datadog:
+    enabled: ${self:custom.staged.dd_enabled, true}
+```
 
 ## Opening Issues
 

--- a/README.md
+++ b/README.md
@@ -35,7 +35,7 @@ To further configure your plugin, use the following custom parameters in your `s
 | `forwarder`          | Setting this parameter subscribes the Lambda functions' CloudWatch log groups to the given Datadog forwarder Lambda function. Required when `enableDDTracing` is set to `true`.                                                                                                                                                                                                                 |
 | `enableTags`         | When set, automatically tag the Lambda functions with the `service` and `env` tags using the `service` and `stage` values from the serverless application definition. It does NOT override if a `service` or `env` tag already exists. Defaults to `true`.                                                                                                                                      |
 | `injectLogContext`         | When set, the lambda layer will automatically patch console.log with Datadog's tracing ids. Defaults to `true`.                                                                                                                                      |
-| `enabled`            | When set to false, the DataDog plugin will stay inactive. Defaults to `true`. |
+| `enabled`            | When set to false, the DataDog plugin will stay inactive. Defaults to `true`. You can control this option using an environment variable, e.g. `enabled: ${strToBool(${env:DD_PLUGIN_ENABLED, true})}`, to activate/deactivate the plugin during deployment. |
 
 
 To use any of these parameters, add a `custom` > `datadog` section to your `serverless.yml` similar to this example:

--- a/README.md
+++ b/README.md
@@ -78,7 +78,7 @@ If you use TypeScript, you may encounter the error of missing type definitions. 
 If using `serverless-webpack`, make sure to also exclude `datadog-lambda-js` and `dd-trace` in your `serverless.yml` in addition to declaring them as external in your webpack config file.
 
 **webpack.config.js**
-```
+```javascript
 var nodeExternals = require('webpack-node-externals')
 
 module.exports = {
@@ -89,7 +89,7 @@ module.exports = {
 ```
 
 **serverless.yml**
-```
+```yaml
 custom:
   webpack:
     includeModules:
@@ -112,7 +112,7 @@ An error occurred: GetaccountapiLogGroupSubscription - Could not execute the lam
 
 If you'd like to turn off the plugin based on the environment (passed via `--stage`), you can use something similar to the example below.
 
-```
+```yaml
 provider:
   stage: ${self:opt.stage, 'dev'}
 

--- a/README.md
+++ b/README.md
@@ -35,6 +35,8 @@ To further configure your plugin, use the following custom parameters in your `s
 | `forwarder`          | Setting this parameter subscribes the Lambda functions' CloudWatch log groups to the given Datadog forwarder Lambda function. Required when `enableDDTracing` is set to `true`.                                                                                                                                                                                                                 |
 | `enableTags`         | When set, automatically tag the Lambda functions with the `service` and `env` tags using the `service` and `stage` values from the serverless application definition. It does NOT override if a `service` or `env` tag already exists. Defaults to `true`.                                                                                                                                      |
 | `injectLogContext`         | When set, the lambda layer will automatically patch console.log with Datadog's tracing ids. Defaults to `true`.                                                                                                                                      |
+| `enabled`            | When set to false, the DataDog plugin will stay inactive. Defaults to `true`. |
+
 
 To use any of these parameters, add a `custom` > `datadog` section to your `serverless.yml` similar to this example:
 

--- a/src/env.ts
+++ b/src/env.ts
@@ -9,6 +9,8 @@
 import Service from "serverless/classes/Service";
 
 export interface Configuration {
+  // Whether Datadog is enabled. Defaults to true.
+  enabled?: boolean;
   // Whether to add the lambda layers, or expect the user's to bring their own
   addLayers: boolean;
   // Datadog API Key, only necessary when using metrics without log forwarding

--- a/src/index.ts
+++ b/src/index.ts
@@ -41,7 +41,7 @@ module.exports = class ServerlessPlugin {
     "before:deploy:function:packageFunction": this.beforePackageFunction.bind(this),
     "before:offline:start:init": this.beforePackageFunction.bind(this),
     "before:step-functions-offline:start": this.beforePackageFunction.bind(this),
-    "after:deploy:deploy": printOutputs.bind(null, this.serverless),
+    "after:deploy:deploy": this.afterDeploy.bind(this),
   };
 
   public commands = {
@@ -63,8 +63,9 @@ module.exports = class ServerlessPlugin {
   constructor(private serverless: Serverless, _: Serverless.Options) {}
 
   private async beforePackageFunction() {
-    this.serverless.cli.log("Auto instrumenting functions with Datadog");
     const config = getConfig(this.serverless.service);
+    if (config.enabled === false) return;
+    this.serverless.cli.log("Auto instrumenting functions with Datadog");
     setEnvConfiguration(config, this.serverless.service);
 
     const defaultRuntime = this.serverless.service.provider.runtime;
@@ -94,6 +95,7 @@ module.exports = class ServerlessPlugin {
 
   private async afterPackageFunction() {
     const config = getConfig(this.serverless.service);
+    if (config.enabled === false) return;
     if (config.forwarder) {
       const aws = this.serverless.getProvider("aws");
       const errors = await addCloudWatchForwarderSubscriptions(this.serverless.service, aws, config.forwarder);
@@ -114,6 +116,12 @@ module.exports = class ServerlessPlugin {
     redirectHandlers(handlers, config.addLayers);
 
     addOutputLinks(this.serverless, config.site);
+  }
+
+  private async afterDeploy() {
+    const config = getConfig(this.serverless.service);
+    if (config.enabled === false) return;
+    return printOutputs(this.serverless);
   }
 
   private debugLogHandlers(handlers: FunctionInfo[]) {

